### ISSUE TITLE
"--notes" option ignores note tags that are entirely punctuation

### ISSUE THREAD

**Issue by dbrookman** [PyCQA/pylint#5840]
### Bug description

If a note tag specified with the `--notes` option is entirely punctuation, pylint won't report a fixme warning (W0511).

```python
# YES: yes
# ???: no
```

`pylint test.py --notes="YES,???"` will return a fixme warning (W0511) for the first line, but not the second.

### Configuration

```ini
Default
```

### Command used

```shell
pylint test.py --notes="YES,???"
```

### Pylint output

```shell
************* Module test
test.py:1:1: W0511: YES: yes (fixme)
```

### Expected behavior

```
************* Module test
test.py:1:1: W0511: YES: yes (fixme)
test.py:2:1: W0511: ???: no (fixme)
```

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.2 (main, Feb  2 2022, 05:51:25) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

macOS 11.6.1

**Comment by DanielNoord** [comment]
Did a little investigation, this is we're actually converting this option in a regular expression pattern (thereby making it awfully similar to the `notes-rgx` option. Since `?` is a special character in regex this doesn't get picked up. Using `\?\?\?` in either `notes` or `notes-rgx` should work.

**Comment by dbrookman** [comment]
Turned out to be the `\b` ending both regex patterns, which would only match tags if the last character was alphanumeric.

**Comment by Pierre-Sassoulas** [comment]
> (thereby making it awfully similar to the notes-rgx option

I'm keeping the issue open for this reason, it seems we should not treat ``notes`` as a regex.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I think I was wrong. Due to the use of `re.escape` I don't think there are any issues with escape characters and it was indeed the problem @dbrookman found. I think we can close this (after the merge).
